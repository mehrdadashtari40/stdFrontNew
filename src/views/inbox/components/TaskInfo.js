
import React, {Component} from 'react';
import jMoment from 'moment-jalaali';
import { withTranslation } from 'react-i18next';

function TaskInfo(props) {
		let rep = props.taskInfo.DURATION.replace("Hours", "ساعت");
		rep = rep.replace("Minute", "دقیقه");
		rep = rep.replace("Seconds", "ثانیه");

		const regex2 = /([0-9]{4})[-](0[1-9]|1[0-2])[-]([0-2]{1}[0-9]{1}) ([0-9]{2}):([0-9]{2}):([0-9]{2})/g;
		let initDate = null;
		if(regex2.exec(props.taskInfo.INIT_DATE)) {
			// true regex date format
			initDate = jMoment(props.taskInfo.INIT_DATE).format('HH:mm:ss jYYYY/jMM/jDD');
		} else {
			// false regex
			initDate = props.t('Case Not Started');
		}


		return (
			<div>
				<table
					className="table table-bordered">
					<tbody>
					<tr>
						<td> {props.t('Title')}</td>
						<td> {props.taskInfo.TAS_TITLE ? props.taskInfo.TAS_TITLE : props.t('Not Value')} </td>
					</tr>
					<tr>
						<td> {props.t('Description')}</td>
						<td>{props.taskInfo.TAS_DESCRIPTION ? props.taskInfo.TAS_DESCRIPTION : props.t('Not Value')}</td>
					</tr>
					<tr>
						<td>{props.t('Init Date')}</td>
						<td>{initDate}</td>
						{/*<td>{jMoment(props.taskInfo.INIT_DATE).format('HH:mm:ss jYYYY/jMM/jDD')}</td>*/}
					</tr>
					<tr>
						<td> {props.t('Due Date')}</td>
						<td>{jMoment(props.taskInfo.DUE_DATE).format('HH:mm:ss jYYYY/jMM/jDD')}</td>
					</tr>
					<tr>
						<td> {props.t('Finish Date')}</td>
						<td>{props.taskInfo.FINISH == "Not finished" ? props.t('Not finished') : jMoment(props.taskInfo.FINISH).format('HH:mm:ss jYYYY/jMM/jDD')}</td>
					</tr>
					<tr>
						<td> {props.t('Task Duration')}</td>
						<td> {props.taskInfo.DURATION == "Not finished" ? props.t('Not finished') : rep }</td>
					</tr>
					</tbody>
				</table>
			</div>
		);
}

export default withTranslation()(TaskInfo);