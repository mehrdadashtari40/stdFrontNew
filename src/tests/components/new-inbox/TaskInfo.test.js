import React from 'react';
import { shallow } from 'enzyme';
import TaskInfo from '../../../views/inbox/components/TaskInfo';

/* Start of SnapShot Test */

jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));


test('Make Snapshot file , TaskInfo Component', () => {
	let fakeTaskInfo = {
		PRO_UID: '1271994115e57596c953e45081760466',
		TAS_UID: '3232592545e575a11045055091238715',
		TAS_ID: 235,
		TAS_TITLE: 'Task 2',
		TAS_DESCRIPTION: '',
		TAS_DEF_TITLE: '',
		TAS_DEF_SUBJECT_MESSAGE: '',
		TAS_DEF_PROC_CODE: '',
		TAS_DEF_MESSAGE: '',
		TAS_DEF_DESCRIPTION: '',
		TAS_TYPE: 'NORMAL',
		TAS_DURATION: 1,
		TAS_DELAY_TYPE: '',
		TAS_TEMPORIZER: 0,
		TAS_TYPE_DAY: '',
		TAS_TIMEUNIT: 'DAYS',
		TAS_ALERT: 'FALSE',
		TAS_PRIORITY_VARIABLE: '',
		TAS_ASSIGN_TYPE: 'SELF_SERVICE',
		TAS_ASSIGN_VARIABLE: '@@SYS_NEXT_USER_TO_BE_ASSIGNED',
		TAS_GROUP_VARIABLE: '',
		TAS_MI_INSTANCE_VARIABLE: '@@SYS_VAR_TOTAL_INSTANCE',
		TAS_MI_COMPLETE_VARIABLE: '@@SYS_VAR_TOTAL_INSTANCES_COMPLETE',
		TAS_ASSIGN_LOCATION: 'FALSE',
		TAS_ASSIGN_LOCATION_ADHOC: 'FALSE',
		TAS_TRANSFER_FLY: 'FALSE',
		TAS_LAST_ASSIGNED: '0',
		TAS_USER: '0',
		TAS_CAN_UPLOAD: 'FALSE',
		TAS_VIEW_UPLOAD: 'FALSE',
		TAS_VIEW_ADDITIONAL_DOCUMENTATION: 'FALSE',
		TAS_CAN_CANCEL: 'FALSE',
		TAS_OWNER_APP: 'FALSE',
		STG_UID: '',
		TAS_CAN_PAUSE: 'FALSE',
		TAS_CAN_SEND_MESSAGE: 'TRUE',
		TAS_CAN_DELETE_DOCS: 'FALSE',
		TAS_SELF_SERVICE: 'FALSE',
		TAS_START: 'FALSE',
		TAS_TO_LAST_USER: 'FALSE',
		TAS_SEND_LAST_EMAIL: 'FALSE',
		TAS_DERIVATION: 'NORMAL',
		TAS_POSX: 247,
		TAS_POSY: 71,
		TAS_WIDTH: 110,
		TAS_HEIGHT: 60,
		TAS_COLOR: '',
		TAS_EVN_UID: '',
		TAS_BOUNDARY: '',
		TAS_DERIVATION_SCREEN_TPL: '',
		TAS_SELFSERVICE_TIMEOUT: 0,
		TAS_SELFSERVICE_TIME: 0,
		TAS_SELFSERVICE_TIME_UNIT: '',
		TAS_SELFSERVICE_TRIGGER_UID: '',
		TAS_SELFSERVICE_EXECUTION: 'EVERY_TIME',
		TAS_NOT_EMAIL_FROM_FORMAT: 0,
		TAS_OFFLINE: 'FALSE',
		TAS_EMAIL_SERVER_UID: '',
		TAS_AUTO_ROOT: 'FALSE',
		TAS_RECEIVE_SERVER_UID: '',
		TAS_RECEIVE_LAST_EMAIL: 'FALSE',
		TAS_RECEIVE_EMAIL_FROM_FORMAT: 0,
		TAS_RECEIVE_MESSAGE_TYPE: 'text',
		TAS_RECEIVE_MESSAGE_TEMPLATE: 'alert_message.html',
		TAS_RECEIVE_SUBJECT_MESSAGE: null,
		TAS_RECEIVE_MESSAGE: null,
		INIT_DATE: '2020-02-27 14:29:38',
		DUE_DATE: '2020-02-28 14:29:30',
		FINISH: 'Not finished',
		DURATION: 'Not finished',
	};
	const wrapper = shallow(<TaskInfo taskInfo={fakeTaskInfo} />);
	expect(wrapper).toMatchSnapshot();
});
