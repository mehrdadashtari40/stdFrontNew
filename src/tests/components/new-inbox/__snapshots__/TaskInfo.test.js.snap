// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , TaskInfo Component 1`] = `
<div>
  <table
    className="table table-bordered"
  >
    <tbody>
      <tr>
        <td>
           
        </td>
        <td>
           
          Task 2
           
        </td>
      </tr>
      <tr>
        <td>
           
        </td>
        <td />
      </tr>
      <tr>
        <td />
        <td>
          14:29:38 1398/12/08
        </td>
      </tr>
      <tr>
        <td>
           
        </td>
        <td>
          14:29:30 1398/12/09
        </td>
      </tr>
      <tr>
        <td>
           
        </td>
        <td />
      </tr>
      <tr>
        <td>
           
        </td>
        <td>
           
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
